set.go

```diff
@@ -76,6 +76,49 @@ type SetElement struct {
 	IntervalEnd bool
 }
 
+func (s *SetElement) decode() func(b []byte) error {
+	return func(b []byte) error {
+		ad, err := netlink.NewAttributeDecoder(b)
+		if err != nil {
+			return fmt.Errorf("failed to create nested attribute decoder: %v", err)
+		}
+		ad.ByteOrder = binary.BigEndian
+
+		for ad.Next() {
+			switch ad.Type() {
+			case unix.NFTA_SET_ELEM_KEY:
+				s.Key, err = decode(ad.Bytes())
+				if err != nil {
+					return err
+				}
+			case unix.NFTA_SET_ELEM_DATA:
+				s.Val, err = decode(ad.Bytes())
+				if err != nil {
+					return err
+				}
+			case unix.NFTA_SET_ELEM_FLAGS:
+				flags := ad.Uint32()
+				s.IntervalEnd = (flags & unix.NFT_SET_ELEM_INTERVAL_END) != 0
+			}
+		}
+		return ad.Err()
+	}
+}
+
+func decode(d []byte) ([]byte, error) {
+
+	ad, err := netlink.NewAttributeDecoder(d)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create nested attribute decoder: %v", err)
+	}
+	ad.ByteOrder = binary.BigEndian
+	var b []byte
+	for ad.Next() {
+		b = ad.Bytes()
```

Comment:
You should check the type of the attribute, and check the error after the loop here.

Reply:
done